# Data Analysis with Python
## ODI_cricket_Bowlers_Analysis
<br>Md. Tawhidul Islam Bhuiyan
<br>Bsc 4th year, Rajshahi University of Engineering & Technology.

### Import required libraries &  Reading xlsx file

In [1]:
import numpy as np
import pandas as pd

In [2]:
df = pd.read_excel("ODI_cricket.xlsx", engine='openpyxl' , sheet_name='bowler')

### First 10 rows of dataset

In [3]:
display(df.head(10))

Player       Span  Mat  Inns  Balls   Runs  Wkts  \
0  M Muralitharan (Asia/ICC/SL)  1993-2011  350   341  18811  12326   534   
1             Wasim Akram (PAK)  1984-2003  356   351  18186  11812   502   
2            Waqar Younis (PAK)  1989-2003  262   258  12698   9919   416   
3          WPUJC Vaas (Asia/SL)  1994-2008  322   320  15775  11014   400   
4  Shahid Afridi (Asia/ICC/PAK)  1996-2015  398   372  17670  13632   395   
5       SM Pollock (Afr/ICC/SA)  1996-2008  303   297  15712   9631   393   
6          GD McGrath (AUS/ICC)  1993-2007  250   248  12970   8391   381   
7                   B Lee (AUS)  2000-2012  221   217  11185   8877   380   
8               SL Malinga (SL)  2004-2019  226   220  10936   9760   338   
9         A Kumble (Asia/INDIA)  1990-2007  271   265  14496  10412   337   

     Ave  Econ    SR   4   5  
0  23.08  3.93  35.2  15  10  
1  23.52  3.89  36.2  17   6  
2  23.84  4.68  30.5  14  13  
3  27.53  4.18  39.4   9   4  
4  34.51  4.62  44.7   4   9  
5  24.50  3.67  39.9  12   5  
6  22.02  3.88  34.0   9   7  
7  23.36  4.76  29.4  14   9  
8  28.87  5.35  32.3  11   8  
9  30.89  4.30  43.0   8   2

#### The Feature variables(columns) of the ODI bowlers dataset :
**Player**: The name of the players.
<br>**Span**: The period of career time of the players.
<br>**Mat**: The number of ODI matches. 
<br>**Inns**: The number of innings.
<br>**Balls**: TThe number of balls bowled. 
<br>**Runs**: The number of runs conceded.
<br>**Wkts**: The number of wickets taken.
<br>**Ave**: The average number of runs conceded per wicket.
<br>**Econ**: The average number of runs conceded per over. (Econ = Runs/Overs bowled). 
<br>**SR**: The average number of balls bowled per wicket taken. (SR = Balls/W).
<br>**4**: The number of innings in which the bowler took at least four wickets.
<br>**5**: The number of matches in which the bowler took at least five wickets.


In [4]:
# number of rows
print("number of rows = ", df.shape[0])

# number of columns
print("number of columns = ", df.shape[1])

number of rows =  77
number of columns =  12


### Features in the dataset and search for missing values and data types

In [5]:
display(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 77 entries, 0 to 76
Data columns (total 12 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Player  77 non-null     object 
 1   Span    77 non-null     object 
 2   Mat     77 non-null     int64  
 3   Inns    77 non-null     int64  
 4   Balls   77 non-null     int64  
 5   Runs    77 non-null     int64  
 6   Wkts    77 non-null     int64  
 7   Ave     77 non-null     float64
 8   Econ    77 non-null     float64
 9   SR      77 non-null     float64
 10  4       77 non-null     int64  
 11  5       77 non-null     int64  
dtypes: float64(3), int64(7), object(2)
memory usage: 7.3+ KB


None

### Staistical Analysis of the dataset

In [6]:
display(df.describe())

Mat        Inns         Balls          Runs        Wkts  \
count   77.000000   77.000000     77.000000     77.000000   77.000000   
mean   194.402597  181.194805   8839.402597   6671.714286  233.805195   
std     82.485606   67.958393   3316.055457   2245.839029   84.406603   
min     80.000000   76.000000   4074.000000   2821.000000  151.000000   
25%    136.000000  128.000000   6182.000000   5058.000000  173.000000   
50%    170.000000  164.000000   8054.000000   6192.000000  199.000000   
75%    227.000000  218.000000  10750.000000   8021.000000  272.000000   
max    463.000000  372.000000  18811.000000  13632.000000  534.000000   

             Ave       Econ         SR          4          5  
count  77.000000  77.000000  77.000000  77.000000  77.000000  
mean   28.958052   4.596753  37.909091   6.350649   2.870130  
std     4.826768   0.515814   6.060901   3.556929   2.530606  
min    18.680000   3.300000  26.100000   1.000000   0.000000  
25%    24.970000   4.280000  33.000000   3.000000   1.000000  
50%    29.290000   4.660000  37.800000   6.000000   2.000000  
75%    31.900000   4.920000  41.400000   8.000000   4.000000  
max    44.480000   5.830000  52.500000  17.000000  13.000000

In [7]:
# renameing columns in dataframe
df = df.rename(columns={'Span' : 'Carrer_Duration',
                        'Mat' : 'Match',
                        'Inns': 'Innings',
                        'Wkts' : 'Wickets',
                        'Ave' : 'Bowling_Average',
                        'Econ': 'Economy_Rate',
                        'SR' : 'Strike_Rate',
                        4: '4_Wickets',
                        5: '5_Wickets'})
display(df.head(10))

Player Carrer_Duration  Match  Innings  Balls   Runs  \
0  M Muralitharan (Asia/ICC/SL)       1993-2011    350      341  18811  12326   
1             Wasim Akram (PAK)       1984-2003    356      351  18186  11812   
2            Waqar Younis (PAK)       1989-2003    262      258  12698   9919   
3          WPUJC Vaas (Asia/SL)       1994-2008    322      320  15775  11014   
4  Shahid Afridi (Asia/ICC/PAK)       1996-2015    398      372  17670  13632   
5       SM Pollock (Afr/ICC/SA)       1996-2008    303      297  15712   9631   
6          GD McGrath (AUS/ICC)       1993-2007    250      248  12970   8391   
7                   B Lee (AUS)       2000-2012    221      217  11185   8877   
8               SL Malinga (SL)       2004-2019    226      220  10936   9760   
9         A Kumble (Asia/INDIA)       1990-2007    271      265  14496  10412   

   Wickets  Bowling_Average  Economy_Rate  Strike_Rate  4_Wickets  5_Wickets  
0      534            23.08          3.93         35.2         15         10  
1      502            23.52          3.89         36.2         17          6  
2      416            23.84          4.68         30.5         14         13  
3      400            27.53          4.18         39.4          9          4  
4      395            34.51          4.62         44.7          4          9  
5      393            24.50          3.67         39.9         12          5  
6      381            22.02          3.88         34.0          9          7  
7      380            23.36          4.76         29.4         14          9  
8      338            28.87          5.35         32.3         11          8  
9      337            30.89          4.30         43.0          8          2

### Splitting Player and Country from 'Player' Column

In [8]:
df['Player'] = df['Player'].str.replace(")"," ")  
df_player = df['Player'].str.split("(", expand=True)
df = df.drop('Player', axis=1)  
df = pd.concat([df, df_player], axis=1) 
df = df.rename(columns={0: 'Player',1: 'Country'})  
column_sequence = ['Player' , 'Country' , 'Carrer_Duration' , 'Match' , 'Innings', 'Balls', 'Runs' , 'Wickets', 'Bowling_Average' , 'Economy_Rate' , 'Strike_Rate' , '4_Wickets' , '5_Wickets']

df = df[column_sequence]

display(df.head())


<ipython-input-8-6e7433b30406>:1: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will*not* be treated as literal strings when regex=True.
  df['Player'] = df['Player'].str.replace(")"," ")


Player        Country Carrer_Duration  Match  Innings  Balls  \
0  M Muralitharan    Asia/ICC/SL        1993-2011    350      341  18811   
1     Wasim Akram            PAK        1984-2003    356      351  18186   
2    Waqar Younis            PAK        1989-2003    262      258  12698   
3      WPUJC Vaas        Asia/SL        1994-2008    322      320  15775   
4   Shahid Afridi   Asia/ICC/PAK        1996-2015    398      372  17670   

    Runs  Wickets  Bowling_Average  Economy_Rate  Strike_Rate  4_Wickets  \
0  12326      534            23.08          3.93         35.2         15   
1  11812      502            23.52          3.89         36.2         17   
2   9919      416            23.84          4.68         30.5         14   
3  11014      400            27.53          4.18         39.4          9   
4  13632      395            34.51          4.62         44.7          4   

   5_Wickets  
0         10  
1          6  
2         13  
3          4  
4          9

**Checking how many players played for ICC**

In [9]:
# filter function 
def ICC(x):
    if "ICC" in x:
        return "Yes"
    else:
        return "No"

# using filter function to check
df['Played_for_ICC'] = df['Country'].apply(ICC)

# removing "ICC/" from string of country column
df['Country'] = df['Country'].str.replace("ICC/","", regex=True)
display(df.head(5))

print(df['Played_for_ICC'].value_counts())

Player    Country Carrer_Duration  Match  Innings  Balls   Runs  \
0  M Muralitharan    Asia/SL        1993-2011    350      341  18811  12326   
1     Wasim Akram        PAK        1984-2003    356      351  18186  11812   
2    Waqar Younis        PAK        1989-2003    262      258  12698   9919   
3      WPUJC Vaas    Asia/SL        1994-2008    322      320  15775  11014   
4   Shahid Afridi   Asia/PAK        1996-2015    398      372  17670  13632   

   Wickets  Bowling_Average  Economy_Rate  Strike_Rate  4_Wickets  5_Wickets  \
0      534            23.08          3.93         35.2         15         10   
1      502            23.52          3.89         36.2         17          6   
2      416            23.84          4.68         30.5         14         13   
3      400            27.53          4.18         39.4          9          4   
4      395            34.51          4.62         44.7          4          9   

  Played_for_ICC  
0            Yes  
1             No  
2             No  
3             No  
4            Yes

No     64
Yes    13
Name: Played_for_ICC, dtype: int64


**Only 13 players played for ICC**

### Checking how many different countries are present in the dataset

In [10]:
print(df['Country'].value_counts())

INDIA          9
PAK            9
AUS            8
NZ             8
WI             6
SL             5
Asia/INDIA     4
Asia/SL        4
SA             4
ENG            4
Afr/SA         4
Asia/PAK       3
BAN            2
AUS/ICC        2
ENG/ICC        2
AFG            1
Afr/ZIM        1
Asia/BAN       1
Name: Country, dtype: int64


There are eight Australian Bowlers and yes one Bangladeshi player in the dataset. 

### Finding the player that has a lowest economy rate

In [11]:
display(df.sort_values(by='Economy_Rate', ascending = True).head(1))

Player Country Carrer_Duration  Match  Innings  Balls  Runs  Wickets  \
65  RJ Hadlee      NZ        1973-1990    115      112   6182  3407      158   

    Bowling_Average  Economy_Rate  Strike_Rate  4_Wickets  5_Wickets  \
65            21.56           3.3         39.1          1          5   

   Played_for_ICC  
65             No


- RJ Hadlee has the lowest strike rate of 3.3

In [12]:
display(df.sort_values(by='Strike_Rate', ascending = True).head(1))

Player Country Carrer_Duration  Match  Innings  Balls  Runs  Wickets  \
41  MA Starc     AUS        2010-2021     99       99   5099  4379      195   

    Bowling_Average  Economy_Rate  Strike_Rate  4_Wickets  5_Wickets  \
41            22.45          5.15         26.1         11          8   

   Played_for_ICC  
41             No

 MA Starc has the lowest strike rate of 26.1

### Finding the player that has a lowest bowling average

In [13]:
display(df.sort_values(by='Bowling_Average', ascending = True).head(1))

Player Country Carrer_Duration  Match  Innings  Balls  Runs  \
74  Rashid Khan     AFG        2015-2022     80       76   4074  2821   

    Wickets  Bowling_Average  Economy_Rate  Strike_Rate  4_Wickets  5_Wickets  \
74      151            18.68          4.15         26.9          5          4   

   Played_for_ICC  
74             No

- Rashid Khan has the lowest bowling average of 18.68